### PR TITLE
Fix SendAsync_ExpectedDiagnosticSourceActivityLogging test

### PR DESCRIPTION
**Pull request by lmolkova**
Fix for SendAsync_ExpectedDiagnosticSourceActivityLogging failure #19689

Behavior of HttpClient diagnostics instrumentation was slightly changed in #19659: 'new' and 'deprecated' events are no longer mutually exclusive.

This change fixes failed test (to reject deprecated events) and adds new test for the case when consumers want both: old and new events.

### CODE REVIEW

**Comment by lmolkova** [comment]
@dotnet-bot build and test outerloop Windows_NT Release x64

**Comment by lmolkova** [comment]
@dotnet-bot test Outerloop Windows_NT Release x64 Build and Test please

**Comment by stephentoub** [comment]
@dotnet-bot test Outerloop netcoreapp Windows_NT Release x64 Build and Test please

**Comment by Priya91** [comment]
@dotnet-bot test outerloop Fedora24 Release x64 Build and Test please

**Comment by Priya91** [comment]
@dotnet-bot test outerloop netcoreapp Fedora24 Release
@dotnet-bot test outerloop netcoreapp Fedora24 Debug
@dotnet-bot test outerloop netcoreapp RHEL7.2 Release
@dotnet-bot test outerloop netcoreapp RHEL7.2 Debug
@dotnet-bot test outerloop netcoreapp CentOS7.1 Release
@dotnet-bot test outerloop netcoreapp CentOS7.1 Debug

**Comment by Priya91** [comment]
Thanks @lmolkova !

**Comment by lmolkova** [comment]
now HttpClientHandler_SslProtocols_Test.GetAsync_SupportedSSLVersion_Succeeds test fails, tracked in #7812

**Comment by stephentoub** [comment]
@dotnet-bot test outerloop netcoreapp CentOS7.1 Release
@dotnet-bot test outerloop netcoreapp CentOS7.1 Debug
@dotnet-bot test Outerloop netcoreapp Windows_NT Release x64 Build and Test please

**Comment by stephentoub** [comment]
@dotnet-bot test OuterLoop netcoreapp Windows_NT Release x64 please

File: src/System.Net.Http/tests/FunctionalTests/DiagnosticsTests.cs [hunk 1]
```diff
@@ -467,6 +466,45 @@ public void SendAsync_ExpectedDiagnosticExceptionActivityLogging()
...
+                    else if (kvp.Key.Equals("System.Net.Http.HttpRequestOut.Start")) { activityStartLogged = true;}
+                    else if (kvp.Key.Equals("System.Net.Http.HttpRequestOut.Stop")) { activityStopLogged = true; }
+                });
+
+                using (DiagnosticListener.AllListeners.Subscribe(diagnosticListenerObserver))
+                {
+                    diagnosticListenerObserver.Enable();
+                    using (var client = new HttpClient())
+                    {
+                        var response = client.GetAsync(Configuration.Http.RemoteEchoServer).Result;
```
Comment (stephentoub) [inline src/System.Net.Http/tests/FunctionalTests/DiagnosticsTests.cs]: Nit: we should Dispose of the response message

**Comment by stephentoub** [comment]
@dotnet-bot test OuterLoop netcoreapp Windows_NT Release x64

**Comment by stephentoub** [comment]
@dotnet-bot test Innerloop netfx Windows_NT Debug x86 Build and Test please (https://github.com/dotnet/corefx/issues/19848)

**Comment by stephentoub** [comment]
@dotnet-bot test OuterLoop netcoreapp Windows_NT Release x64 (certificate validation errors?)

File: src/System.Net.Http/tests/FunctionalTests/DiagnosticsTests.cs [hunk 2]
```diff
@@ -488,7 +488,7 @@ public void SendAsync_ExpectedDiagnosticSourceNewAndDeprecatedEventsLogging()
                     diagnosticListenerObserver.Enable();
                     using (var client = new HttpClient())
                     {
-                        var response = client.GetAsync(Configuration.Http.RemoteEchoServer).Result;
+                        client.GetAsync(Configuration.Http.RemoteEchoServer).Result.Dispose();
```
Comment (lmolkova) [inline src/System.Net.Http/tests/FunctionalTests/DiagnosticsTests.cs]: thanks!

There are other tests in the same class that never dispose response. I can fix other tests

**Pull request merged**